python/ray/data/dataset.py

```diff
@@ -317,88 +318,124 @@ def map_batches(
         fn_constructor_kwargs: Optional[Dict[str, Any]] = None,
         **ray_remote_args,
     ) -> "Dataset[Any]":
-        """Apply the given function to batches of records of this dataset.
+        """Apply the given function to batches of data.
 
-        The format of the data batch provided to ``fn`` can be controlled via the
-        ``batch_format`` argument, and the output of the UDF can be any batch type.
+        Batches are represented as dataframes, arrays, or lists. The default batch
+        type is determined by your dataset's schema. To determine the default batch
+        type, call :meth:`~Dataset.native_batch_format`. Alternatively, set the batch
+        type with ``batch_format``.
 
-        This is a blocking operation.
+        To learn more about writing functions for :meth:`~Dataset.map_batches`, read
+        :ref:`writing user-defined functions <transform_datasets_writing_udfs>`.
+
+        .. note::
+            This is a blocking operation. The time complexity is :math:`O(n/p)`, where
+            :math:`n` is the dataset size and :math:`p` is the parallelism.
+
+        .. tip::
+            Consider using :class:`~ray.data.preprocessors.BatchMapper`. It's more
```

Comment:
Might want to qualify this with "If using Ray AIR for training and/or batch inference,"